core/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -204,23 +209,35 @@ private static long optionalSum(long left, long right) {
                     }
                 }
             }
-            return scoreDocs;
+            final boolean isSorted;
+            final SortField[] sortFields;
+            if (mergedTopDocs != null && mergedTopDocs instanceof TopFieldDocs) {
+                TopFieldDocs fieldDocs = (TopFieldDocs) mergedTopDocs;
+                isSorted = (fieldDocs instanceof CollapseTopFieldDocs &&
+                    fieldDocs.fields.length == 1 && fieldDocs.fields[0].getType() == SortField.Type.SCORE) == false;
+                sortFields = fieldDocs.fields;
+            } else {
+                isSorted = false;
+                sortFields = null;
+            }
+            return new SortedTopDocs(scoreDocs, isSorted, sortFields);
         } else {
-            // no relevant docs - just return an empty array
-            return EMPTY_DOCS;
+            // no relevant docs
+            return SortedTopDocs.EMPTY;
         }
     }
 
-    private ScoreDoc[] mergeTopDocs(Collection<TopDocs> results, int topN, int from) {
+    TopDocs mergeTopDocs(Collection<TopDocs> results, int topN, int from) {
         if (results.isEmpty()) {
-            return EMPTY_DOCS;
+            return null;
```

Comment:
because the return value is again `TopDocs` and not `ScoreDocs[]` so `    private static final ScoreDoc[] EMPTY_DOCS = new ScoreDoc[0];` wouldn't cut it